src/pal/inc/mbusafecrt.h

```diff
@@ -56,7 +56,7 @@ extern WCHAR* wcstok_s( WCHAR* inString, const WCHAR* inControl, WCHAR** ioConte
 // strnlen is not required unless the source string is completely untrusted (e.g. anonymous input on a website)
 #ifndef SUPPRESS_STRNLEN
     extern size_t PAL_strnlen( const char* inString, size_t inMaxSize );
-    extern size_t wcsnlen( const WCHAR* inString, size_t inMaxSize );
+    extern size_t PAL_wcsnlen( const WCHAR* inString, size_t inMaxSize );
```

Comment:
You are missing two additional changes:
In pal.h, next to `#define strnlen       PAL_strnlen` you need to add `#define wcsnlen       PAL_wcsnlen`
In palinternal.h, next to `#undef strnlen` you need to add `#undef wcsnlen`. 
Otherwise wcsnlen would not be visible in the coreclr sources.